app/api/auth/[...nextauth]/options.ts

```diff
@@ -1,9 +1,22 @@
 import { RedisClientType, createClient } from "falkordb";
 import CredentialsProvider from "next-auth/providers/credentials"
-import { AuthOptions } from "next-auth"
+import { AuthOptions, User } from "next-auth"
 
 
 export const connections = new Map<number, RedisClientType>();
+
+export async function getConnection(user: User) {
+    return connections.get(user.id) ?? await createClient({
```

Comment:
@Anchel123 notice if you create a new connection you should store it in the connections otherwise each call you won't find the connection in the map and create a new connection